docs/config.md

```diff
@@ -318,6 +319,110 @@ location /blog/ {
 
 Default: `[]`
 
+- `NoDNSLink`
+  A boolean to configure whether DNSLink lookup for value in `Host` HTTP header should be performed.
+  If DNSLink is present, content path stored in it is mounted at `/` and respective payload is returned to the client.
+
+  Default: `false`
+
+### Gateway.PublicGateways
+
+`PublicGateways` is a dictionary for overriding default options on specific fully qualified domain names (FQDN).  
+Options for changing behavior per hostname:
+
+- `NoDNSLink`  
+  Overrides global setting of the same name.  
+  Default: `false`
+- `PathPrefixes`  
+  Array of paths that should be mounted at the root of domain name.  
+  **TODO: this does not override global 'PathPrefixes' used for validation of X-Ipfs-Gateway-Prefix header. Should this have different name? eg. PathMounts?**
+  Example: `["/ipfs", "/ipns", "/api"]`  
+  Default: `[]`
+  - **Note:** when both `PathPrefixes` and DNSLink are mounted, `PathPrefixes` take priority
+- `UseSubdomains`
+  A boolean to configure whether the gateway provides [Origin isolation](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy)
+  between content roots.  
+  Default: `false`
+  - `true` – mount [subdomain gateway](#https://docs-beta.ipfs.io/how-to/address-ipfs-on-web/#subdomain-gateway) at `http://{cid}.ipfs.{hostname}`
+    - requests for `http://{hostname}/ipfs/{cid}` will return redirect to `http://{cid}.ipfs.{hostname}`
+  - `false` – mount [path gateway](https://docs-beta.ipfs.io/how-to/address-ipfs-on-web/#path-gateway) at `http://{hostname}/ipfs/{cid}`
+    - **(not implemented yet)** due to the lack of Origin isolation, cookies and storage will be disabled by [Clear-Site-Data](https://github.com/ipfs/in-web-browsers/issues/157) header
```

Comment:
Nice!

Reply:
I think I'll drop it from the scope fo this PR and add `Clear-Site-Data` in a separate one as it may require adding new flag (as noted in https://github.com/ipfs/in-web-browsers/issues/157#issuecomment-580341836)